examples/features/name_resolving/client/main.go

```diff
@@ -0,0 +1,122 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"google.golang.org/grpc"
+	ecpb "google.golang.org/grpc/examples/features/proto/echo"
+	"google.golang.org/grpc/resolver"
+)
+
+const (
+	exampleScheme      = "example"
+	exampleServiceName = "resolver.example.grpc.io"
+
+	backendAddr = "localhost:50051"
+)
+
+var addrs = []string{backendAddr}
+
+func callUnaryEcho(c ecpb.EchoClient, message string) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	r, err := c.UnaryEcho(ctx, &ecpb.EchoRequest{Message: message})
+	if err != nil {
+		log.Fatalf("could not greet: %v", err)
+	}
+	fmt.Println(r.Message)
+}
+
+func makeRPCs(cc *grpc.ClientConn, n int) {
+	hwc := ecpb.NewEchoClient(cc)
+	for i := 0; i < n; i++ {
+		callUnaryEcho(hwc, "this is examples/name_resolving")
+	}
+}
+
+func main() {
+	passthroughConn, err := grpc.Dial(
+		fmt.Sprintf("passthrough:///%s", backendAddr), // Dial to "passthrough:///localhost:50051"
+		grpc.WithInsecure(),
+	)
+	if err != nil {
+		log.Fatalf("did not connect: %v", err)
+	}
+	defer passthroughConn.Close()
+
+	fmt.Println("--- calling helloworld.Greeter/SayHello with pick_first ---")
+	makeRPCs(passthroughConn, 10)
+
+	fmt.Println()
+
+	exampleConn, err := grpc.Dial(
+		fmt.Sprintf("%s:///%s", exampleScheme, exampleServiceName), // Dial to "example:///resolver.example.grpc.io"
+		grpc.WithInsecure(),
+	)
+	if err != nil {
+		log.Fatalf("did not connect: %v", err)
+	}
+	defer exampleConn.Close()
+
+	fmt.Println("--- calling helloworld.Greeter/SayHello with pick_first ---")
+	makeRPCs(exampleConn, 10)
+}
+
+// Following is an example name resolver implementation.
```

Comment:
Add a link to the resolver builder and resolver interface? 
And either in the readme or in the code, describe that user need to implement the resolver builder interface and the resolver interface to have a complete grpc resolver component implementation?